3570/1/top_gems.rb

```diff
@@ -0,0 +1,60 @@
+require 'bundler'
+require 'optparse'
+require 'yaml'
+Bundler.require(:default)
+require_relative 'outputer_basic'
+require_relative 'parser'
+
+options = {}
+
+OptionParser.new do |opts|
+  opts.banner = 'Usage: top_gems.rb [options]'
+
+  opts.on('--top=NUM') do |num|
+    options[:top] = num.to_i
+  end
+
+  opts.on('--name=NAME') do |name|
+    options[:name] = name
+  end
+
+  opts.on('--file=PATH') do |path|
+    options[:gems] = YAML.load_file(path)['gems']
+  end
+end.parse!
+
+options[:gems] ||= YAML.load_file('gems_list.yml')['gems']
+
+class TopGems
+  def initialize(options)
+    @gem_name = options[:name]
+    @top = options[:top]
+    @gems = options[:gems]
+  end
+
+  def run
+    gems_info = Parser.new(@gems).parse
+    result = sort_gems(gems_info)
+    result = get_gem_for_name(result) if @gem_name
+    result = get_gems_for_top(result) if @top
+    Outputer.new(result).output_in_table
+  end
+
+  private
+
+  def sort_gems(gems_info)
+    (gems_info.sort_by { |_gem_name, info| info[:stars] }).reverse
```

Comment:
parentheses are not needed here. As well as `reverse` btw. Jus add `-` before ur block return value